### PR TITLE
fix: Reading invalid formatted plifile should raise error instead of warning

### PR DESCRIPTION
**Pull request by MRVermeulenDeltares**
Refs: #320

### CODE REVIEW

File: hydrolib/core/dflowfm/polyfile/parser.py [hunk 1]
```diff
@@ -358,15 +380,15 @@ def _add_current_block_as_incomplete_error(self) -> None:
...
+        msg.notify_as_error(self._file_path)
 
     def _parse_name_or_new_description(self, line: str) -> None:
         if Parser._is_comment(line):
             self._handle_new_description(line)
         elif Parser._is_name(line):
             self._handle_parse_name(line)
         elif self._state != StateType.INVALID_STATE:
-            self._handle_new_error("Expected a valid name or description")
+            self._handle_new_error("Settings of block might be incorrect, expected a valid name or description")
```
[the remark this answers is not in the document]
Comment (tim-vd-aardweg) [reply hydrolib/core/dflowfm/polyfile/parser.py]: Discussed. This is fine!

**Review by tim-vd-aardweg — CHANGES_REQUESTED** [review]

File: tests/dflowfm/test_polyfile.py [hunk 2]
```diff
@@ -616,30 +651,35 @@ def test_empty_lines_is_correctly_logged(
                 ),
                 [((0, 3), "Expected a valid next point at line 3.")],
             ),
+            (
+                inspect.cleandoc(
+                    """
+                    *description
+                    name
+                    1  5   
+                    1.0 2.0 3.0 4.0 5.0 Comment after the values is invalid"""
```
[the remark this answers is not in the document]
Comment (MRVermeulenDeltares) [reply tests/dflowfm/test_polyfile.py:670]: Added

File: hydrolib/core/dflowfm/polyfile/parser.py [hunk 3]
```diff
@@ -51,7 +51,29 @@ def notify_as_warning(self, file_path: Optional[Path] = None):
...
+        """Call warnings.warn with a formatted string describing this ParseMsg
+
+        Args:
+            file_path (Optional[Path], optional):
+                The file path mentioned in the warning if specified. Defaults to None.
+        """
+        warning_message = self._format_parsemsg_to_string(file_path)
+        warnings.warn(warning_message)
+
+    def notify_as_error(self, file_path: Optional[Path] = None):
```
[the remark this answers is not in the document]
Comment (MRVermeulenDeltares) [reply hydrolib/core/dflowfm/polyfile/parser.py]: adjusted

File: hydrolib/core/dflowfm/polyfile/parser.py [hunk 4]
```diff
@@ -460,8 +440,13 @@ def _handle_new_error(self, reason: str) -> None:
...
 
-    def _handle_parse_msg(self, msg: ParseMsg) -> None:
-        msg.notify_as_warning(self._file_path)
+    def _notify_as_warning(self, msg: ParseMsg) -> None:
+        warning_message = msg.format_parsemsg_to_string(self._file_path)
+        warnings.warn(warning_message)
+
+    def _notify_as_error(self, msg: ParseMsg) -> None:
+        error_message = msg.format_parsemsg_to_string(self._file_path)
+        raise ValueError(f"Invalid formatted plifile, {error_message}")
```
Comment (tim-vd-aardweg) [inline hydrolib/core/dflowfm/polyfile/parser.py:449]: Add to documentation that we can raise a `ValueError`

**Review by tim-vd-aardweg — CHANGES_REQUESTED** [review]

**Review by tim-vd-aardweg — APPROVED** [review]

**Pull request merged**